v3/pkg/application/linux_cgo.go

```diff
@@ -1010,17 +1010,41 @@ func onDragNDrop(target unsafe.Pointer, context *C.GdkDragContext, x C.gint, y C
 
 //export onKeyPressEvent
 func onKeyPressEvent(widget *C.GtkWidget, event *C.GdkEventKey, userData unsafe.Pointer) C.gboolean {
-	windowId := uint(*((*C.uint)(userData)))
-	fmt.Println("onKeyPressEvent", windowId)
-	/*
-		windowKeyEvents <- &windowKeyEvent{
+    accelerator := getKeyboardState(event)
+	windowKeyEvents <- &windowKeyEvent{
 			windowId:          windowID,
-			acceleratorString: C.GoString(acceleratorString),
-		}
-	*/
+			acceleratorString: accelerator,
+	}	
 	return C.gboolean(0)
 }
 
+func getKeyboardState(event *C.GdkEventKey) string {
+    modifiers := uint(event.state) & C.GDK_MODIFIER_MASK
+    keyCode := uint(event.keyval)
+
+    var acc accelerator
+    // Check Accelerators
+    if modifiers&(C.GDK_SHIFT_MASK) != 0 {
+        acc.Modifiers = append(acc.Modifiers, ShiftKey)
+    }
+    if modifiers&(C.GDK_CONTROL_MASK) != 0 {
+        acc.Modifiers = append(acc.Modifiers, ControlKey)
+    }
+    if modifiers&(C.GDK_MOD1_MASK) != 0 {
+        acc.Modifiers = append(acc.Modifiers, OptionOrAltKey)
+    }
+    if modifiers&(C.GDK_SUPER_MASK) != 0 {
+        acc.Modifiers = append(acc.Modifiers, SuperKey)
+    }
+    keyString, ok := VirtualKeyCodes[keyCode]
+    if !ok {
+        fmt.Println("No Matching Keycode found for key:", keyCode)
+    }
+    acc.Key = keyString
+	string := acc.String()
+    return string
```

Comment:
as mentioned above I think it would be better to return a `(string, bool)` at this point considering you are doing a `fmt.Println` on line `1041` if it fails to find the keycode.  I am unsure if the accelerator string is even valid at this point if there was no matching keycode.